src/openms/source/QC/FragmentMassError.cpp

```diff
@@ -0,0 +1,269 @@
+      }
+      if (exp_spectrum.getMSLevel() != 2)
+      {
+        throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "The matching retention time of the mzML is not a MS2 Spectrum.");
+      }
+
+      //-----------------------------------------------------------------------
+      // COMPARE THEORETICAL AND EXPERIMENTAL SPECTRUM
+      //-----------------------------------------------------------------------
+
+      if (exp_spectrum.empty() || theo_spectrum.empty())
+      {
+        LOG_WARN << "The spectrum with " + std::to_string(exp_spectrum.getRT()) + " is empty." << "\n";
+        return;
+      }
+
+      //stores ppms for one spectrum
+      DoubleList ppms{};
+
+      //infinity
+      double inf = std::numeric_limits<double>::infinity();
+
+      //exp_peak matching to previous theo_peak
+      double current_exp = inf;
+
+      //minimal ppm
+      double ppm = inf;
+
+      for (const Peak1D& peak : theo_spectrum)
+      {
+        const double theo_mz = peak.getMZ();
+        Size index = exp_spectrum.findNearest(theo_mz);
+        const double exp_mz = exp_spectrum[index].getMZ();
+
+        const double mz_tolerance = tolerance_unit_ppm ?  theo_mz * tolerance * 1e-6 : tolerance;
+
+
+        //found peak match
+        if (std::abs(theo_mz-exp_mz) < mz_tolerance)
+        {
+          auto current_ppm = Math::getPPM(theo_mz, exp_mz);
+
+          //first peak in tolerance range
+          if (current_exp == inf)
+          {
+            ppm = current_ppm;
+            current_exp = exp_mz;
+          }
+
+          //theo_peak matches to a exp_peak that is already matched
+          //&& ppm is smaller than before
+          if (current_exp == exp_mz && abs(current_ppm) < abs(ppm))
+          {
+            ppm = current_ppm;
+          }
+
+          //theo_peak matches to another exp_peaks
+          if (current_exp != exp_mz)
+          {
+            ppms.push_back(ppm);
+
+            ++ counter_ppm;
+
+            accumulator_ppm += ppm;
+            ppm = current_ppm;
+            current_exp = exp_mz;
+          }
+
+         }
+      }
+
+      //last peak doesn't have a successor so it has to be added manually
+      ppms.push_back(ppm);
+      accumulator_ppm += ppm;
+      ++ counter_ppm;
+
+
+      //-----------------------------------------------------------------------
+      // WRITE PPM ERROR IN PEPTIDEHIT
+      //-----------------------------------------------------------------------
+
+      pep_id.getHits()[0].setMetaValue("ppm_errors", ppms);
+
+    };
+
+    auto lamVar = [&result, &counter_ppm](const PeptideIdentification& pep_id)
+    {
+      for (auto ppm : (pep_id.getHits()[0].getMetaValue("ppm_errors")).toDoubleList())
+      {
+        result.variance_ppm += pow((ppm - result.average_ppm),2);
+      }
+    };
+
+    //computation of ppms
+    QCBase::iterateFeatureMap(fmap, lamCompPPM);
```

Comment:
ouch!
this lambda copies the WHOLE MSExperiment each time its called (which is very often!).
Copying the MSExperiment is not necessary here. Just find the correct spectrum, copy the spectrum, mow it and compare.
All the Mower initalization code should also only be executed once.
If you find your lambda to have more than a few lines of code, its not a good lambda anymore. Write a proper private function (with documentation...) or do not use a function at all (which is hard in this case because PepIDs are in multiple places).